### PR TITLE
Tensile-style block to C tile map

### PR DESCRIPTION
**Pull request by rosenrodt**
This PR depends on #235. See [diff against #235](
https://github.com/ROCmSoftwarePlatform/composable_kernel/compare/rosenrodt/block-to-ctile-refactor...rosenrodt/new-ctile-map)

- Add 2 more tile map variants. M01"Adapt" means the bottom-most M01-by-N0 group can shrink on M01 dimension so that all mapped index is within C matrix boundary. So any "adapted" C tile map instance can apply to all problem sizes.
  - `BlockToCTileMap_M00_N0_M01Adapt` for gemm and batched gemm
  - `BlockToCTileMap_KSplit_M00_N0_M01Adapt` for split-k gemm
- Apply new tile map to all XDL gridwise gemm implementations
- Quick benchmark on MI100 @1087mhz shows gemm_xdl_fp16 improves from 113 Tflops to 116 Tflops (7680 x 8192 x 8192)

### CODE REVIEW

**Comment by zjing14** [comment]
@rosenrodt please resolve conflicts

**Comment by rosenrodt** [comment]
@zjing14 Rebased. This PR now depends on #246 hot fix

File: include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp
Comment (zjing14) [inline include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp:223]: @rosenrodt Is `M01 = 8 ` empirical?

Comment (rosenrodt) [reply include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp:223]: Historically MI60 has 64 CUs so that'll make M01_N0 an 8x8 tile group since the idea is to make the group as square as possible. For MI100 forwards `M01=8` is still pretty good based on empirical findings on Tensile. Maybe Henry (@aazz44ss) can comment on that

Comment (aazz44ss) [reply include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp:223]: agree, M01=8 is default on Tensile.

**Review by zjing14 — APPROVED** [review]
Looks good to me for now. But we need to open an "after-merge" issue to figure out the cause of branches in  `CalculateBottomIndex` with transforms and get ride of raw index calculation.

**Comment by zjing14** [comment]
Hold this PR until #246 merged

File: include/ck/utility/block_to_ctile_map.hpp
Comment (asroy) [reply include/ck/utility/block_to_ctile_map.hpp]: I see this comment and code are removed. There is no longer inefficient ISA issue?

**Comment by rosenrodt** [comment]
@zjing14 @asroy 

I have removed dead code as requested. Please review.

**Review by zjing14 — APPROVED** [review]
LGTM

**Review by asroy — APPROVED** [review]

**Pull request merged**
